src/optimizer/optimizer_custom_rules.cpp

```diff
@@ -0,0 +1,16 @@
+#include "execution/plans/abstract_plan.h"
+#include "optimizer/optimizer.h"
+
+namespace bustub {
+
+auto Optimizer::OptimizeCustom(const AbstractPlanNodeRef &plan) -> AbstractPlanNodeRef {
+  auto p = plan;
+  p = OptimizeMergeProjection(p);
+  p = OptimizeMergeFilterNLJ(p);
+  p = OptimizeNLJAsIndexJoin(p);
+  // p = OptimizeNLJAsHashJoin(p);  // Enable this rule after you have implemented hash join.
+  p = OptimizeOrderByAsIndexScan(p);
```

Comment:
I can add in top N related stuff together with the SQL tests in another PR

Reply:
Yes, I'll help modify your PR and merge.